javatools/build.gradle.kts

```diff
@@ -98,6 +101,23 @@ val jar by tasks.existing(Jar::class) {
     }
 }
 
+/**
+ * TODO: Marcus suggests that merging the jars is not a good way to go. Instead we simply need to
+ *       get the jline.jar into our distribution (installDist task) and point to it in the manifest
+ *       of the javatools.jar
+ */
+val shadowJar by tasks.existing(ShadowJar::class) {
+    dependsOn(jar)
+
+    archiveBaseName = "javatools"
+
+    from(tasks.jar.get().archiveFile)
+
```

Comment:
You know you are building multiple jars, right? The shadow jar gets a classifier "all" if you take a look at the build results. This is not the javatools.jar that ends up part of the installDist, but rather the one without jline as a dependency. I have taken a look inside the "all" jar, and it seems to contain only the jline classes without overly transitive dependencies, which is good, but I don't think it works as a dependency for the platform right now, if the distribution ships the standard one. This is partly why I wanted to avoid the shadowJar. I guess we have to make other modules aware of the twain jar artifacts somehow. The classifier is probably fine, maven wise, if you publish a maven component it will default to also building a "source" classifier zip file and a "javadoc" classifier zip file, but I'm not sure what the preferred way of shipping the javatools jar with the all classifier, and what needs to be changed in its dependencies. The xtcJavaTools configuration probably resolves the standard jar still, and other dependencies on it. The plugin looks for a javatools.jar on its classpath, and is probably bundle with the javatools.jar.

Reply:
Yes, I know that. To test, I currently make the shadow and copy the '-all' jar to the install dir manually:
`./gradlew :javatools:shadowJar;cp /Users/ggleyzer/Development/xvm/javatools/build/libs/javatools-0.4.43-all.jar /Users/ggleyzer/Development/xvm/xdk/build/install/xdk/javatools/javatools.jar`